panic in "istioctl manifest apply"

To reproduce:

```
cat > icp.yaml <<EOF
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
metadata:
  namespace: istio-operator
  name: example-istiocontrolplane
spec:
EOF
istioctl manifest apply -f icp.yaml
```

I discovered this testing https://preliminary.istio.io/docs/reference/config/istio.operator.v1alpha12.pb/ which suggests the default Istio istall can be achieved with a bare `spec:`

Someone should verify if this backtraces the operator container.

The panic is at 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x88 pc=0x296f2a6]

goroutine 1 [running]:
istio.io/istio/operator/pkg/validate.CheckIstioOperatorSpec(0x0, 0xc00044b500, 0x0, 0x0, 0x6)
	istio.io/istio@/operator/pkg/validate/validate.go:41 +0x26
```

--- a/operator/cmd/mesh/manifest-generate_test.go
+++ b/operator/cmd/mesh/manifest-generate_test.go
@@ -233,6 +233,17 @@ func TestMultiICPSFiles(t *testing.T) {
 	})
 }
 
+func TestBareSpec(t *testing.T) {
+	testDataDir = filepath.Join(repoRootDir, "cmd/mesh/testdata/manifest-generate")
+	t.Run("multi-ICPS files", func(t *testing.T) {
+		inPathBase := filepath.Join(testDataDir, "input/bare_spec.yaml")
+		_, err := runManifestGenerate([]string{inPathBase}, "")
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
 // TestLDFlags checks whether building mesh command with
 // -ldflags "-X istio.io/pkg/version.buildHub=myhub -X istio.io/pkg/version.buildVersion=mytag"
 // results in these values showing up in a generated manifest.

--- a/operator/cmd/mesh/profile-common.go
+++ b/operator/cmd/mesh/profile-common.go
@@ -100,7 +100,7 @@ func parseYAMLFiles(inFilenames []string, force bool, l *Logger) (overlayYAML st
 		}
 		l.logAndErrorf("Validation errors (continuing because of --force):\n%s", err)
 	}
-	if fileOverlayIOP.Spec.Profile != "" {
+	if fileOverlayIOP.Spec != nil && fileOverlayIOP.Spec.Profile != "" {
 		if profile != "" && profile != fileOverlayIOP.Spec.Profile {
 			return "", "", fmt.Errorf("different profiles cannot be overlaid")
 		}

--- a/operator/cmd/mesh/testdata/manifest-generate/input/bare_spec.yaml
+++ b/operator/cmd/mesh/testdata/manifest-generate/input/bare_spec.yaml
@@ -1,0 +1,3 @@
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:

--- a/operator/pkg/controlplane/control_plane.go
+++ b/operator/pkg/controlplane/control_plane.go
@@ -50,15 +50,17 @@ func NewIstioOperator(installSpec *v1alpha1.IstioOperatorSpec, translator *trans
 		out.components = append(out.components, component.NewCoreComponent(c, &o))
 	}
 
-	for idx, c := range installSpec.Components.IngressGateways {
-		o := *opts
-		o.Namespace = defaultIfEmpty(c.Namespace, iop.Namespace(installSpec))
-		out.components = append(out.components, component.NewIngressComponent(c.Name, idx, c, &o))
-	}
-	for idx, c := range installSpec.Components.EgressGateways {
-		o := *opts
-		o.Namespace = defaultIfEmpty(c.Namespace, iop.Namespace(installSpec))
-		out.components = append(out.components, component.NewEgressComponent(c.Name, idx, c, &o))
+	if installSpec.Components != nil {
+		for idx, c := range installSpec.Components.IngressGateways {
+			o := *opts
+			o.Namespace = defaultIfEmpty(c.Namespace, iop.Namespace(installSpec))
+			out.components = append(out.components, component.NewIngressComponent(c.Name, idx, c, &o))
+		}
+		for idx, c := range installSpec.Components.EgressGateways {
+			o := *opts
+			o.Namespace = defaultIfEmpty(c.Namespace, iop.Namespace(installSpec))
+			out.components = append(out.components, component.NewEgressComponent(c.Name, idx, c, &o))
+		}
 	}
 	for _, cn := range orderedKeys(installSpec.AddonComponents) {
 		c := installSpec.AddonComponents[cn]

--- a/operator/pkg/validate/validate.go
+++ b/operator/pkg/validate/validate.go
@@ -38,6 +38,10 @@ var (
 // call the appropriate validation function. checkRequiredFields determines whether missing mandatory fields generate
 // errors.
 func CheckIstioOperatorSpec(is *v1alpha1.IstioOperatorSpec, checkRequiredFields bool) (errs util.Errors) {
+	if is == nil {
+		return util.Errors{}
+	}
+
 	errs = CheckValues(is.Values)
 	return util.AppendErrs(errs, validate(defaultValidations, is, nil, checkRequiredFields))
 }

